src/main/infrastructure/repository/iab/GVLFactory.js

```diff
@@ -1,36 +1,24 @@
 import {GVL} from '@iabtcf/core'
+import {
+  VENDOR_LIST_DEFAULT_LANGUAGE,
+  VENDOR_LIST_LATEST_VERSION
+} from '../../../core/constants'
 
 export class GVLFactory {
-  constructor({baseUrl = 'https://a.dcdn.es/borostcf/v2/vendorlist'} = {}) {
+  constructor({
+    baseUrl = 'https://a.dcdn.es/borostcf/v2/vendorlist',
+    language = VENDOR_LIST_DEFAULT_LANGUAGE
+  } = {}) {
     GVL.baseUrl = baseUrl
-    this._setStaticGVLParameters({language: 'en'})
-    GVL.languageFilename = 'LATEST?language=[LANG]'
-  }
-
-  create({version} = {}) {
-    return new GVL(version)
-  }
-
-  async createWith({version, language} = {}) {
-    const gvl = this.create({version})
-    await this._changeLanguage({language, gvl})
-    return gvl
-  }
-
-  resetCaches() {
-    GVL.emptyCache()
-    GVL.emptyLanguageCache()
+    GVL.latestFilename = `${VENDOR_LIST_LATEST_VERSION}?language=${language}`
+    GVL.versionedFilename = `[VERSION]?language=${language}`
+    this._cached = new Map()
   }
 
-  async _changeLanguage({language, gvl}) {
-    if (language && language !== gvl.language) {
-      this._setStaticGVLParameters({language})
-      await gvl.changeLanguage(language)
+  create({version = VENDOR_LIST_LATEST_VERSION} = {}) {
+    if (!this._cached.has(version)) {
+      this._cached.set(version, new GVL(version))
```

Comment:
GVL is created in english by default https://github.com/InteractiveAdvertisingBureau/iabtcf-es/blob/6c3562ab0189918f3563e876a1246de1e9b05a36/modules/core/src/GVL.ts#L244
Internally you are created in English but you are setting in spanish the url

Reply:
Correct, with this PR, we'll not use the "language" used in the GVL, but we'll use the language given from the outside to request the appropriate language to our service URL